src/shape.cpp

```diff
@@ -420,6 +420,20 @@ void shape::multi_copy(std::size_t idx, std::size_t* start, const std::size_t* e
     *start = tidx;
 }
 
+bool shape::check_within_bounds(std::vector<std::size_t> multi) const
```

Comment:
I would just name this `within_bounds`.

Reply:
Renamed to multi_within_bounds to address your and Lakinder's comments.